src/md.cc

```diff
@@ -429,10 +452,35 @@ void MGmol<OrbitalsType>::md(OrbitalsType** orbitals, Ions& ions)
         int retval              = 0;
         bool small_move         = true;
         bool last_move_is_small = true;
-        do
+
+#ifdef MGMOL_HAS_LIBROM
+        // variables in ROM MVP solver for Pinned H2O
+        std::vector<double> positions;
+        std::vector<short> anumbers;
+        PinnedH2O H2O_molecule;
+        if (ct.getROMOptions().rom_stage == ROMStage::ONLINE_PINNED_H2O_3DOF)
+        {
+            getAtomicPositions(positions);
+            getAtomicNumbers(anumbers);
+            H2O_molecule.rotate(positions, anumbers);
+            ions.setPositions(positions, anumbers);
+            Potentials& pot = hamiltonian_->potential();
+            pot.initialize(ions);
+            //moveVnuc(ions);
```

Comment:
I think we need to add here:     g_kbpsi_->setup(ions);
or maybe better at the beginning of "updateDMandEnergy()"
This setup another part of the potential (the non-local one)

Reply:
This addition gives an error `error: invalid use of incomplete type ‘using element_type = class KBPsiMatrixSparse’ {aka ‘class KBPsiMatrixSparse’}`

Reply:
You need to add #include "KBPsiMatrixSparse.h" in this file